### DISCUSSION TITLE
How can i reload the change of DL Models on Fastapi?

### DISCUSSION THREAD

**Discussion by mennatallah644** [fastapi/fastapi#8701] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
no code example
```

### Description

When i change the files of models the models files does not seem to be reloaded and the old ones seems to be used , so how can i make the change in models appear automatically also i dont want to reload the whole app using reload=True is there a way to do this ?

### Operating System

Linux

### FastAPI Version

0.70.0

### Python Version

3.7

**Comment by raphaelauv** [comment]
only reload=True will work

**Comment by Kludex** [comment]
You can use `--reload-include` and `--reload-exclude`. Available on `uvicorn>=0.15.0`.

**Comment by HGamalElDiin** [comment]
Thanks really @Kludex, appreciate your help.

one more thing please, how to get benefit of using the reload with workers?

**Comment by Kludex** [comment]
You don't. Reload on uvicorn runs on single process.

**Comment by HGamalElDiin** [comment]
@Kludex 
Noway to reload the app after a certain condition without using "reload" and take the advantage of workers?

**Comment by Kludex** [comment]
Not that I am aware.

**Comment by insomnes** [comment, marked as answer]
You can create asyncio task on startup (and cancel it on shutdown) which checks files hashsum and sleeps for some time, if hashsum has been changed you reload your model into memory.
```python3
import asyncio
from hashlib import md5
from fastapi import FastAPI

FILE_PATH = "test.txt"


class MemLoader:
    def __init__(self, file_path: str):
        self.file_path = file_path
        self.file_content, self.file_hash = self.load_file()

    def load_file(self):
        with open(self.file_path) as file:
            file_content = file.read()

        file_hash = md5(file_content.encode("UTF-8")).digest()

        return file_content, file_hash

    def check_file_changed(self):
        file_content, file_hash = self.load_file()
        if file_hash != self.file_hash:
            print("FILE CHANGED RELOAD")
            self.file_content = file_content
            self.file_hash = file_hash

    def get_content(self):
        return self.file_content


app = FastAPI()


async def check_file_changed(mem_loader: MemLoader):
    loop = asyncio.get_running_loop()
    print("STARTING FILE CHECK")
    while True:
        await asyncio.sleep(15)
        await loop.run_in_executor(
            executor=None, func=mem_loader.check_file_changed
        )


@app.on_event("startup")
async def load_mem_loader():
    print("LOADING MEM LOADER")
    app.mem_loader = MemLoader(FILE_PATH)
    app.mem_loader_task = asyncio.create_task(
        check_file_changed(app.mem_loader)
    )


@app.on_event("shutdown")
async def cancel_mem_loader():
    app.mem_loader_task.cancel()
    print("CHECK TASK CANCELED")



@app.get("/content")
async def get_content():
    return app.mem_loader.get_content()
```

**Comment by HGamalElDiin** [comment]
@insomnes 

This worked as test thanks really..
but i'm wondering is there an optimized way to use asyncio to track a group of files?
like given an extension regex "*.pkl"?

Appreciate your response.

**Comment by insomnes** [comment]
@HGamalElDiin you can change value in 
```python
await asyncio.sleep(15)
```
For sleep timer
> but i'm wondering is there an optimized way to use asyncio to track a group of files?
like given an extension regex "*.pkl"?

You can write it yourself. I've showed you the concept of running some background task for it with asyncio. 
You can run any code like that. If you can't write the code for multiple files you can ask for help at Stackoverflow.

**Comment by HGamalElDiin** [comment]
@insomnes thanks for your help really, i did that and woks fine now

**Comment by insomnes** [comment]
@HGamalElDiin if it works for you, can you please close issue?
